builtin/k8s/platform.go

```diff
@@ -125,6 +128,165 @@ func (p *Platform) getClientset() (*clientsetInfo, error) {
 	}, nil
 }
 
+func (p *Platform) resourceDeploymentStatus(
+	ctx context.Context,
+	log hclog.Logger,
+	sg terminal.StepGroup,
+	deploymentState *Resource_Deployment,
+	clientset *clientsetInfo,
+	sr *resource.StatusResponse,
+) error {
+	s := sg.Add("Checking status of the kubernetes deployment resource...")
+	defer s.Abort()
+
+	namespace := p.config.Namespace
+	if namespace == "" {
+		namespace = clientset.Namespace
+	}
+
+	// Get deployment status
+
+	deploymentResource := sdk.StatusReport_Resource{
+		Platform:            platformName,
+		Type:                "deployment",
+		CategoryDisplayHint: sdk.ResourceCategoryDisplayHint_INSTANCE_MANAGER,
+	}
+	sr.Resources = append(sr.Resources, &deploymentResource)
+
+	deployResp, err := clientset.Clientset.AppsV1().Deployments(namespace).Get(ctx, deploymentState.Name, metav1.GetOptions{})
+	if deployResp == nil {
+		return status.Errorf(codes.FailedPrecondition, "kubernetes deployment response cannot be nil")
+	} else if err != nil {
+		if !errors.IsNotFound(err) {
+			return status.Errorf(codes.FailedPrecondition, "error getting kubernetes deployment %s: %s", deploymentState.Name, err)
+		} else {
+			deploymentResource.Name = deploymentState.Name
+			deploymentResource.Health = sdk.StatusReport_MISSING
+			deploymentResource.HealthMessage = sdk.StatusReport_MISSING.String()
+
+			// Continue on with getting statuses for the rest of our resources
+		}
+	} else {
+		// Found the deployment, and can use it to populate our resource
+
+		var mostRecentCondition v1.DeploymentCondition
+		for _, condition := range deployResp.Status.Conditions {
+			if condition.LastUpdateTime.Time.After(mostRecentCondition.LastUpdateTime.Time) {
```

Comment:
Nit: I think we can remove the `.Time` here; `LastUpdateTime` is a `metav1.Time` which wraps Go's `Time` 

-https://github.com/kubernetes/apimachinery/blob/1af25b613b6482b465c4bf23501a9b02acdb3c0c/pkg/apis/meta/v1/time.go#L24-L33

Reply:
Hmm, that doesn't seem to work. It wraps time.Time, but doesn't implement it:

`cannot use mostRecentCondition.LastUpdateTime (type "k8s.io/apimachinery/pkg/apis/meta/v1".Time) as type time.Time in argument to condition.LastUpdateTime.Time.After`

Reply:
> as type time.Time in argument to condition.LastUpdateTime.Time.After

What about removing both `.Time`'s: 

```go
if condition.LastUpdateTime.After(mostRecentCondition.LastUpdateTime) {
```

but it's just a nit so don't worry too much about it

Reply:
Well interesting, I can remove the first .Time, but not the second 😅 . I'll go with tacking it on consistently I think.